docs/_posts/2023-04-20-functional-modularity-2.md

```diff
@@ -176,13 +226,51 @@ object Tree {
+
+## Footnotes
+
+[^1]: [listing for ImmutableList in the Scala 2.13 standard library docs](<https://www.scala-lang.org/api/2.13.3/scala/collection/immutable/List.html#map[B](f:A=%3EB):List[B]>)
+[^2]: A few resources to visit for pattern matching: first, the [docs on pattern matching from Scala's website](<https://docs.scala-lang.org/tour/pattern-matching.html>) and then [an article interviewing Martin Odersky on why pattern matching is an important feature](<https://www.artima.com/articles/the-point-of-pattern-matching-in-scala>)
+[^3]: For more information about generic classes, check out this [official scala guide](<https://docs.scala-lang.org/tour/generic-classes.html>)
+[^4]: Scala compiler's powerful type inference is able to infer from the integers we pass in for each `value` argument that we are making a `Tree[Int]` type tree without us having to indicate it. The compiler would also throw an exception if we were to mix in objects of other types as `value`s for some of these `Nodes`. I left out the type annotation from myTree which would have looked like `val myTree: Tree[Int]`, but it is good practice to include type annotations that are otherwise inferrable for humans reading your code except where you feel it is hurting readability.
+[^5]: The requirement that there be case statements for every type that extends a sealed trait in Scala is termed an "exhaustive matching."
+[^6]: if you opted to work through this before looking at the solution and were stumped, congratulations for putting in that extra effort! You can also take heart from this small anecdote: In Hughes's original publication of this memo, he actually had only one foldTree function to handle both of the foldTree sub-problems outlined in this post. That version as written would have required somehow handling the `Tree[A]` and `List[Tree[A]]` types as though they could be matched as the same type. Over time, due to repeated mentions by the community of readers of his memo that the pseudocode didn't translate into a solution in an actual language (Haskell in most cases), Hughes re-published this article so that `redTree` and `redTree'` were the two functions and they called each other in a very similar fashion to the solution I provided here. When writing this post, I also ended up stuck trying to figure out a way to solve this problem given the constraints of his pseudocode and was unable to adapt my code until I became aware of his updated publication. I posted a link to both the original and updated publication in the first footnote of my [first post](/2023/03/functional-modularity-1/#fn:1)
```

Comment:
```suggestion
[^1]: [listing for ImmutableList in the Scala 2.13 standard library docs](https://www.scala-lang.org/api/2.13.3/scala/collection/immutable/List.html#map[B](f:A=%3EB):List[B])
[^2]: A few resources to visit for pattern matching: first, the [docs on pattern matching from Scala's website](https://docs.scala-lang.org/tour/pattern-matching.html) and then [an article interviewing Martin Odersky on why pattern matching is an important feature](https://www.artima.com/articles/the-point-of-pattern-matching-in-scala)
[^3]: For more information about generic classes, check out this [official scala guide](https://docs.scala-lang.org/tour/generic-classes.html)
[^4]: Scala compiler's powerful type inference is able to infer from the integers we pass in for each `value` argument that we are making a `Tree[Int]` type tree without us having to indicate it. The compiler would also throw an exception if we were to mix in objects of other types as `value`s for some of these `Nodes`. I left out the type annotation from myTree which would have looked like `val myTree: Tree[Int]`, but it is good practice to include type annotations that are otherwise inferrable for humans reading your code except where you feel it is hurting readability.
[^5]: The requirement that there be case statements for every type that extends a sealed trait in Scala is termed an "exhaustive matching."
[^6]: if you opted to work through this before looking at the solution and were stumped, congratulations for putting in that extra effort! You can also take heart from this small anecdote: In Hughes's original publication of this memo, he actually had only one foldTree function to handle both of the foldTree sub-problems outlined in this post. That version as written would have required somehow handling the `Tree[A]` and `List[Tree[A]]` types as though they could be matched as the same type. Over time, due to repeated mentions by the community of readers of his memo that the pseudocode didn't translate into a solution in an actual language (Haskell in most cases), Hughes re-published this article so that `redTree` and `redTree'` were the two functions and they called each other in a very similar fashion to the solution I provided here. When writing this post, I also ended up stuck trying to figure out a way to solve this problem given the constraints of his pseudocode and was unable to adapt my code until I became aware of his updated publication. I posted a link to both the original and updated publication in the first footnote of my [first post](/2023/03/functional-modularity-1/#fn:1)
```